security/ssh_certificate.ksy

```diff
@@ -0,0 +1,196 @@
+        '"ecdsa-sha2-nistp256-cert-v01@openssh.com"': cert_ecdsa
+        '"ecdsa-sha2-nistp384-cert-v01@openssh.com"': cert_ecdsa
+        '"ecdsa-sha2-nistp521-cert-v01@openssh.com"': cert_ecdsa
+        '"ssh-ed25519-cert-v01@openssh.com"': cert_ed25519
+types:
+  cert_rsa:
+    doc-ref: 'https://tools.ietf.org/html/rfc4253#section-6.6'
+    seq:
+      - id: nonce
+        doc: CA-provided random bitstring to prevent hash collisions in the signature
+        type: cstring_bytes
+      - id: rsa
+        type: ssh_public_key::key_rsa
+      - id: footer
+        type: cert_footer
+  cert_dss:
+    seq:
+      - id: nonce
+        doc: CA-provided random bitstring to prevent hash collisions in the signature
+        type: cstring_bytes
+      - id: dsa
+        type: ssh_public_key::key_dsa
+      - id: footer
+        type: cert_footer
+  cert_ecdsa:
+    seq:
+      - id: nonce
+        doc: CA-provided random bitstring to prevent hash collisions in the signature
+        type: cstring_bytes
+      - id: ecdsa
+        type: ssh_public_key::key_ecdsa
+      - id: footer
+        type: cert_footer
+  cert_ed25519:
+    seq:
+      - id: nonce
+        doc: CA-provided random bitstring to prevent hash collisions in the signature
+        type: cstring_bytes
+      - id: ed25519
+        type: ssh_public_key::key_ed25519
+      - id: footer
+        type: cert_footer
+  cert_footer:
+    seq:
+      - id: serial
+        doc: optional serial number; if zero the CA doesn't use serial numbers
+        type: u8
+      - id: type
+        doc: specifies the certificate type (user vs host)
+        type: u4
+        enum: cert_type
+      - id: key_id
+        type: cstring_utf8
+      - id: valid_principals
+        doc: | 
+          As a special case, a zero-length "valid principals"  field means
+          the certificate is valid for any principal of the specified type.
+        type: packed_cstring
+      - id: valid_after
+        doc: Unix timestamp (seconds since 1970-01-01 00:00:00)
+        type: u8
+      - id: valid_before
+        doc: | 
+          Unix timestamp (seconds since 1970-01-01 00:00:00)
+          When "forever" is requested this is set to 0xFFFF_FFFF_FFFF_FFFF
+        type: u8
+      - id: critical_options
+        doc: | 
+          critical options is a set of zero or more key options encoded as
+          below. All such options are "critical" in the sense that an implementation
+          must refuse to authorise a key that has an unrecognised option.
+
+          Generally, critical options are used to control features that restrict
+          access where extensions are used to enable features that grant access.
+          This ensures that certificates containing unknown restrictions do not
+          inadvertently grant access while allowing new protocol features to be
+          enabled via extensions without breaking certificates' backwards
+          compatibility.
```

Comment:
I find this too verbose and confused. If you found time to shorten it and make it clearer, it would be great.